api/src/main/java/com/epam/pipeline/manager/pipeline/PipelineRunManager.java

```diff
@@ -1004,11 +1022,8 @@ private RunInstance configureRunInstance(PipelineConfiguration configuration) {
                 .orElse(instance.getNodeDisk()));
         instance.setNodeType(configuration.getInstanceType());
         instance.setNodeImage(configuration.getInstanceImage());
-        AbstractCloudRegion cloudRegion = Optional.ofNullable(configuration.getCloudRegionId())
-                .map(id -> cloudRegionManager.load(id))
-                .orElse(cloudRegionManager.loadDefaultRegion());
-        instance.setCloudRegionId(cloudRegion.getId());
-        instance.setCloudProvider(cloudRegion.getProvider());
+        Optional.ofNullable(region).map(AbstractCloudRegion::getId).ifPresent(instance::setCloudRegionId);
```

Comment:
`region` must be not null at this point

Reply:
It is true for common runs but for DTS runs it is fine to have no region specified.